### PR TITLE
HttpResponseParser - add parsing for status codes without reason (#1251)

### LINKED ISSUE #1251: HttpResponseParser returns invalid status code for statuses without reason phrase
When server responds with HTTP status without reason phase then it is not parsed as status code, but only skipped, so always default statusCode value (200 OK) is returned.

### PR DESCRIPTION
**Pull request by mateuszgruszczynski**
Fixed #1251 and updated tests to use non-default status code

### CODE REVIEW

**Comment by jlprat** [comment]
OK TO TEST

File: akka-http-core/src/test/scala/akka/http/impl/engine/parsing/ResponseParserSpec.scala [hunk 1]
```diff
@@ -85,12 +85,12 @@ abstract class ResponseParserSpec(mode: String, newLine: String) extends FreeSpe
...
 
       "a response with a missing reason phrase" in new Test {
-        s"HTTP/1.1 200 ${newLine}Content-Length: 0${newLine}${newLine}" should parseTo(HttpResponse(OK))
+        s"HTTP/1.1 404 ${newLine}Content-Length: 0${newLine}${newLine}" should parseTo(HttpResponse(NotFound))
         closeAfterResponseCompletion shouldEqual Seq(false)
       }
 
       "a response with no reason phrase and no trailing space" in new Test {
-        s"""HTTP/1.1 200${newLine}Content-Length: 0${newLine}${newLine}""".stripMargin should parseTo(HEAD, HttpResponse())
+        s"""HTTP/1.1 404${newLine}Content-Length: 0${newLine}${newLine}""".stripMargin should parseTo(HEAD, HttpResponse(NotFound))
```
Comment (ktoso) [inline akka-http-core/src/test/scala/akka/http/impl/engine/parsing/ResponseParserSpec.scala]: Why these changes? Seems unrelated to what the commit wants to fix.

No test for the case the PR claims to fix?

File: akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpResponseParser.scala [hunk 2]
```diff
@@ -108,6 +108,7 @@ private[http] class HttpResponseParser(protected val settings: ParserSettings, p
           maxResponseReasonLength + " characters")
       skipReason(startIdx)
     } else if (isNewLine(cursor + 3)) {
+      parseStatusCode()
```
Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpResponseParser.scala]: I'm not confident this does what you think, see below comment which explains that we actually DO accept such responses (and I remember the PR actually too..)

File: akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpResponseParser.scala [hunk 3]
```diff
@@ -108,6 +108,7 @@ private[http] class HttpResponseParser(protected val settings: ParserSettings, p
...
       // Status format with no reason phrase and no trailing space accepted, diverging from the spec
       // See https://github.com/akka/akka-http/pull/989
```
Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpResponseParser.scala]: AFAICS what you want is implemented and commented/explained here, what's not working about it?

**Comment by mateuszgruszczynski** [comment]
Well, I could chose better name for that PR. Responses were accepted, but since `parseStatusCode()` was not executed in `else if` block, what was done as "parsing" was only skipping chars that contain status code, instead of: 
- reading that code int value
- matching it with predefined StatusCodes 
- updating statusCode variable, which is by default set to `StatusCode.OK`

so all tests that was testing `HTTP 200` status code was always false positive. If it it still not clear what was the problem please just try to run those two tests on master branch, with any code other than 200.

**Comment by ktoso** [comment]
I see, that explains things, would be nice to include such context in initial PR comment :-)
Checked the flow control and can confirm this fixes the issue, thanks a ton! 👍
